Flecsi fails with lapacke errors

On Moonlight, I am getting the following error:
```
/lustre/scratch1/turquoise/certik/repos/flecsi/src/specializations/burton/test/burton_gradients.cc:84:29: fatal error: lapacke/lapacke.h: No such file or directory
```
Which can be fixed by:
```diff
--- a/src/specializations/burton/test/burton_gradients.cc
+++ b/src/specializations/burton/test/burton_gradients.cc
@@ -81,7 +81,7 @@ protected:
 
 // this may not be a portable way to point to lapacke.h. it works on darwin.
 #include <ccomplex>
-#include <lapacke/lapacke.h>
+#include <lapacke.h>
 
 const int N = 10;
 ```
But then I get:
```
[ 18%] Linking CXX executable test/specializations/burton_gradients
/usr/bin/ld: cannot find -llapacke
collect2: error: ld returned 1 exit status
```
which I don't know how to fix, because the `third-party` didn't install any `lapacke.so`:
```
$ ls $TPL_INSTALL_PREFIX/lib
cmake            libhdf5_hl.a      libnetcdf.settings  libscotchmetis.a
libexodus.a      libhdf5.settings  libscotch.a         libszip.a
libexoIIv2for.a  libmetis.a        libscotcherr.a      libz.a
libhdf5.a        libnetcdf.a       libscotcherrexit.a  pkgconfig
```
Here are my copy & paste instructions to build on Moonlight:
```
bash
module load friendly-testing git gcc cmake/3.4.1

export TPL_INSTALL_PREFIX=$HOME/ext/flecsi-libs

git clone https://github.com/flecsi/third-party.git flecsi-thirdparty
cd flecsi-thirdparty
mkdir build
cd build
cmake -DCMAKE_INSTALL_PREFIX=$TPL_INSTALL_PREFIX \
    -DENABLE_EXODUS=ON \
    -DENABLE_LAPACK=ON \
    -DENABLE_LAPACK=ON \
    -DENABLE_METIS=ON \
    -DENABLE_SCOTCH=ON \
    ..
make -j16


git clone --recursive -b beta-0.0-603-ge2dd9992 https://github.com/flecsi/flecsi.git
cd flecsi
mkdir build
cd build
../arch/developer-gnu
make # FAILS as indicated above
make test
```

--- a/config/packages.cmake
+++ b/config/packages.cmake
@@ -176,11 +176,11 @@ if(NOT APPLE)
 
   # This is a workaround that finds the TPL install if it's there. The link
   # to -lgfortan is a serious hack, and a documented issue.
-  set(LAPACK_FOUND)
+  set(LAPACKE_FOUND)
   set(LAPACK_LIBRARIES)
   if(EXISTS ${TPL_INSTALL_PREFIX}/include/lapacke.h
      AND EXISTS ${TPL_INSTALL_PREFIX}/lib64/liblapacke.a)
-    set(LAPACK_FOUND 1)
+    set(LAPACKE_FOUND 1)
     include_directories(${TPL_INSTALL_PREFIX}/include)
     list( APPEND LAPACK_LIBRARIES
           ${TPL_INSTALL_PREFIX}/lib64/liblapacke.a
@@ -190,7 +190,13 @@ if(NOT APPLE)
   else()
     # append lapacke to list of lapack libraries
     find_package(LAPACK)
-    list( APPEND LAPACK_LIBRARIES lapacke)
+    find_library(LAPACKE_LIB NAMES lapacke)
+    find_path(LAPACKE_INCLUDE_DIRS NAMES lapacke.h)
+    if(LAPACKE_INCLUDE_DIRS AND LAPACK_FOUND AND LAPACKE_LIB)
+      set(LAPACKE_FOUND TRUE)
+      include_directories(${LAPACKE_INCLUDE_DIRS})
+      list( APPEND LAPACK_LIBRARIES ${LAPACKE_LIB})
+    endif(LAPACKE_INCLUDE_DIRS AND LAPACK_FOUND AND LAPACKE_LIB)
     # want to add ${LAPACK_INCLUDES}/lapacke to the include search path,
     # but FindLAPACK.cmake defines no such variable.
   endif()

--- a/src/specializations/CMakeLists.txt
+++ b/src/specializations/CMakeLists.txt
@@ -45,12 +45,12 @@ cinch_add_unit(burton
   LIBRARIES flecsi
 )
 
-if(LAPACK_FOUND AND ENABLE_IO)
+if(LAPACKE_FOUND AND ENABLE_IO)
   cinch_add_unit(burton_gradients
     SOURCES burton/test/burton_gradients.cc
     LIBRARIES flecsi ${LAPACK_LIBRARIES} ${IO_LIBRARIES}
   )
-endif(LAPACK_FOUND AND ENABLE_IO)
+endif(LAPACKE_FOUND AND ENABLE_IO)
 
 if(ENABLE_IO)
   cinch_add_unit(burton_io

